syntax = "proto3";

package ticket_reservation;

service TicketReservation{
  // A simple RPC.
  //
  // Obtains the MessageResponse at a given position.
 rpc ReserveTicket(ReservationRequest) returns (ReservationResponse) {}
 rpc ModifyTicket(ReservationRequest) returns (ReservationResponse) {}
 rpc CancelTicket(ReservationRequest) returns (ReservationResponse) {}

}

message user_details{
 string first_name = 1;
 string last_name = 2;
 string email = 3;
 string address = 4;
 uint32 seat = 5;
 string section = 6;
}

message ReservationRequest{
 optional uint64 ticket_no = 1; 
 string from_code = 2;
 string to_code = 3;
 uint64 price_paid = 4;
 uint64 passenger_count = 5;
 repeated user_details passengers = 6;
}

message ReservationResponse{
 uint64 ticket_no = 1;
 string from_code = 2;
 string to_code = 3;
 uint64 price_paid = 4;
 uint64 passenger_count = 5;
 repeated user_details passengers = 6;
 string status = 7;
}